crates/trigger/src/config.rs

```diff
@@ -0,0 +1,32 @@
+use std::path::PathBuf;
+
+use anyhow::Result;
+use serde::Deserialize;
+use toml;
+
+// Config for config providers and wasmtime config
+#[derive(Debug, Default, Deserialize)]
+pub struct TriggerExecutorBuilderConfig {
+    pub vault_config: Option<VaultConfig>,
```

Comment:
I think it would be good for the file structure here to be slightly more generic, allowing for (future) provider plugins and multiple providers of the same type. Perhaps something like:

```toml
[[config_provider]]
type = "vault"
# url = ...
```
> note that order would matter here; Providers are checked in order

This is pretty straightforward with serde: https://serde.rs/enum-representations.html#internally-tagged

I think it would be something like (untested):

```rust
#[derive(Debug, Default, Deserialize)]
pub struct TriggerExecutorBuilderConfig {
    #[serde(rename = "config_provider", default)]
    pub config_providers: Vec<ConfigProvider>,
}

#[derive(Debug, Default, Deserialize)]
#[serde(rename_all = "snake_case", tag = "type")]
pub enum ConfigProvider {
    Vault(VaultConfig),
    // Could add later e.g.: Env(EnvConfig),
}
```